### PR TITLE
Shielding Auras

### PR DESCRIPTION
**Pull request by Geevies**
Ports https://github.com/Baystation12/Baystation12/pull/19992 and https://github.com/Baystation12/Baystation12/pull/27266.

* Added a personal shield device to the traitor tools uplink.
* Added a radiant shielding aura spell to Battlemage and Cleric.
* Added an exosuit shield drone to the mechfab.

This works, but I'm not entirely happy with the sprites. The personal shield has no on-mob shimmer, it uses a mindbatterer grenade as an icon sprite. The wizard radiant spell uses a pretty big and janky sprite as an on-mob, and the exosuit's shield sprite doesn't follow the dir the mech faces, but that one I can maybe fix on my own. If anyone knows about that last one, lemme know, Bay had code for it that we do not.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/objects/auras/auras.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	log_debug("Bullet Act for \ref[src]: [P], [def_zone]")
+	return returning
+
+/obj/aura/debug/life_tick()
+	log_debug("Life tick")
+	return returning
+
+/obj/aura/debug/hitby(atom/movable/M, speed)
+	log_debug("Hit By for \ref[src]: [M], [speed]")
+	return returning
```
Comment (NonQueueingMatt) [inline code/game/objects/auras/auras.dm:55]: I dunno if you wanna keep all these debug logs, it could get very spammy

**Comment by Kaedwuff** [comment]
Is there an example of the sprites being used now for these auras?

File: code/game/objects/auras/auras.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*Auras are simple: They are simple overriders for attackbys, bullet_act, damage procs, etc. They also tick after their respective mob.
+They should be used for undeterminate mob effects, like for instance a toggle-able forcefield, or indestructability as long as you don't move.
+They should also be used for when you want to effect the ENTIRE mob, like having an armor buff or showering candy everytime you walk.
+*/
+
+/obj/aura
+	var/mob/living/user
+
+/obj/aura/New(var/mob/living/target)
```
Comment (skull132) [inline code/game/objects/auras/auras.dm:9]:
```suggestion
/obj/aura/Initialize(var/mob/living/target)
```

File: code/game/objects/auras/auras.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	..()
```
Comment (skull132) [inline code/game/objects/auras/auras.dm:10]:
```suggestion
	. = ..()
```

File: code/game/objects/auras/auras.dm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	return FALSE
+
+/obj/aura/hitby(atom/movable/M, speed)
+	return FALSE
+
+/obj/aura/debug
+	var/returning = 0
+
+/obj/aura/debug/attackby(obj/item/I, mob/user)
+	log_debug("Attackby for \ref[src]: [I], [user]")
```
Comment (skull132) [inline code/game/objects/auras/auras.dm:42]:
```suggestion
```
shitty log.

File: code/game/objects/auras/auras.dm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	return returning
+
+/obj/aura/debug/bullet_act(obj/item/projectile/P, def_zone)
+	log_debug("Bullet Act for \ref[src]: [P], [def_zone]")
```
Comment (skull132) [inline code/game/objects/auras/auras.dm:46]:
```suggestion
```

File: code/game/objects/auras/auras.dm [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	return returning
+
+/obj/aura/debug/life_tick()
+	log_debug("Life tick")
```
Comment (skull132) [inline code/game/objects/auras/auras.dm:50]:
```suggestion
```

File: code/game/objects/auras/auras.dm [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	return returning
+
+/obj/aura/debug/hitby(atom/movable/M, speed)
+	log_debug("Hit By for \ref[src]: [M], [speed]")
```
Comment (skull132) [inline code/game/objects/auras/auras.dm:54]:
```suggestion
```

File: code/game/objects/auras/personal_shield.dm [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+/obj/aura/personal_shield/device
+	var/obj/item/device/personal_shield/shield
+
+/obj/aura/personal_shield/device/bullet_act()
+	. = ..()
+	if(shield)
+		shield.take_charge()
+
+/obj/aura/personal_shield/device/New(var/mob/living/user, var/user_shield)
```
Comment (skull132) [inline code/game/objects/auras/personal_shield.dm:29]: Initialize this.

File: code/game/objects/auras/radiant_aura.dm [hunk 9]
```diff
@@ -0,0 +1,21 @@
+/obj/aura/radiant_aura
+	name = "radiant aura"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "at_shield1"
+	alpha = 75
+	layer = ABOVE_MOB_LAYER
+
+/obj/aura/radiant_aura/New()
```
Comment (skull132) [inline code/game/objects/auras/radiant_aura.dm:8]: Initialize this.

**Pull request merged**